src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,213 @@
+package duke;
+
+import duke.command.DukeException;
+import duke.command.IllegalDateTimeException;
+import duke.command.IllegalDescriptionException;
+import duke.io.ReadFromFile;
+import duke.io.WriteToFile;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.util.ArrayList;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class Duke {
+    final static int LENGTH_TODO = 4;
+    final static int LENGTH_DEADLINE = 8;
+    final static int LENGTH_EVENT = 5;
+    final static int LENGTH_BY = 3;
+    final static int LENGTH_AT = 3;
+    final static int SIZE_LINE = 90;
+
+    enum typeOfTasks{
+        TODO, EVENT, DEADLINE
+    }
+    private static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        String command;
+        printGreeting();
+        // Load saved data if any
+        ReadFromFile.readFromMem(tasks);
+        // Keeps reading and printing user output while input is not bye
+        do {
+            command = in.nextLine();
+            try {
+                getCommandOutput(command);
+            } catch (NoSuchElementException exception) {
+                System.out.println(" Stop feeding me emptiness");
+                printLine();
+            } catch (IndexOutOfBoundsException exception) {
+                System.out.println(" Stop feeding me things that do not exist!");
+                printLine();
+            } catch (DukeException exception) {
+                System.out.println(exception.toString());
+                printLine();
+            }
+        } while (!command.equals("bye"));
+    }
+    public static void printLine() {
+        String dash = "\u2500";
+        System.out.println(dash.repeat(SIZE_LINE));
+    }
+    public static void printInstructions() {
+        System.out.println(" - Add a todo e.g. todo read book");
+        System.out.println(" - Add a deadline e.g. deadline do quiz /by 1st Sept");
+        System.out.println(" - Add an event e.g. event prom /at Mon 6pm");
+        System.out.println(" - List all added tasks e.g. list");
+        System.out.println(" - Complete a task e.g. done 1");
+        System.out.println(" - Delete a task e.g. delete 1");
+        System.out.println(" - Say bye to me e.g. bye");
+    }
+    public static void printGreeting(){
+        printLine();
+        System.out.println(" Hello! I'm Duke");
+        System.out.println(" What can I do for you?");
+        printInstructions();
+        printLine();
+    }
+    public static void printBye(){
+        System.out.println(" Bye. Hope to see you again soon!");
+    }
+
+    // Decides on the output for each command
```

Comment:
Explanation of methods should be done with Javadoc comment rather than inline comment